test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java

```diff
@@ -214,8 +214,9 @@ private boolean isTestMachineryDomain(ProtectionDomain domain) {
         CodeSource cs = (domain == null) ? null : domain.getCodeSource();
         URL loc = (cs == null) ? null : cs.getLocation();
         String path = (loc == null) ? null : loc.getPath();
-        return path != null && TEST_JARS.stream()
-                                .filter(path::endsWith)
+        String name = (path == null) ? null : path.substring(path.lastIndexOf('/') + 1);
```

Comment:
Is there a chance that this path could end with `/` and thus lead to a `IndexOutOfBoundsException` on this call to `substring`?

Reply:
Yes. Usually the URL of the domain's code source points to a JAR file - but it is possible that a directory is the code source. Then the URL could end with a `/` character.

Will refactor the fix to use `Path::getFileName`.